kyo-prelude/shared/src/main/scala/kyo2/Abort.scala

```diff
@@ -5,51 +5,50 @@ import kernel.Const
 import kernel.Effect
 import kernel.Frame
 import kyo.Tag
+import kyo2.Result.*
 import kyo2.kernel.Reducible
 import scala.annotation.targetName
 import scala.reflect.ClassTag
-import scala.util.Failure
-import scala.util.Success
-import scala.util.Try
 
-sealed trait Abort[-E] extends Effect[Const[E], Const[Unit]]
+sealed trait Abort[-E] extends Effect[Const[Failure[E]], Const[Unit]]
 
 object Abort:
 
     given eliminateAbort: Reducible.Eliminable[Abort[Nothing]] with {}
     private inline def erasedTag[E]: Tag[Abort[E]] = Tag[Abort[Any]].asInstanceOf[Tag[Abort[E]]]
 
-    inline def fail[E](inline value: E): Nothing < Abort[E] =
-        Effect.suspendMap[Any](erasedTag[E], value)(_ => ???)
+    inline def error[E](inline value: E): Nothing < Abort[E] =
```

Comment:
I'm renaming this method to follow the naming in `Result`. `Fiber` will also have an `error` method.

Reply:
How would you feel about renaming the method `error` -> `fail` in `Result`? the underlying `Error` type can stay the same.

using the 'verb' for the constructor method:
- `success` -> `succeed`
- `error` -> `fail`
- `panic` -> `panic`

Reply:
I think having different names for the type `Error` and the correspondent constructor `fail` is less intuitive. Another suggestion:

```scala
opaque type Result[+E, +A] >: (Success[A] | Error[E]) = Success[A] | Error[E]

sealed abstract class Error[+E]
case class Fail[+E](error: E) extends Error[E] // or Failure?
case class Panic(exception: Throwable) extends Error[Nothing]

// Other types like Abort and Fiber would also provide methods with the same name:
Result.success(...)
Result.fail(...)
Result.panic(...)
```

Reply:
Btw, what do you think about renaming `Success` to `Value`? It's shorter and matches `result.value: Maybe[A]`. We could also rename `result.value` to `result.success` but I think it's less intuitive.

Reply:
I like your first suggestion. In my opinion, `Value` loses the semantic meaning of a computation with the potential for failure. I think it is also more likely to feel familiar to users of 
`Try`

Reply:
Naming updated